File: 3. Longest Substring Without Repeating Characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        int l = 0, t = 0;
        int i = 0;
        while(i < s.Length){
            blt(s, ref t, i);
            if(i - t + 1 > l)
                l = i - t + 1;
            i++;
        }
        return l;
    }
    public void blt(string s, ref int t, int j){
        for(int i = t; i < j; i++)
            if(s[i] == s[j])
                t = i + 1;
    }
}

File: 3. Longest Substring Without Repeating Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int l = s.length();
        if (l == 0)return 0;
        int[] a = new int[l];
        a[0] = 1;
        int h = 1;
        for(int i = 1; i < l; i++){
            a[i] = 1;
            for (int k = 1; k <= a[i - 1]; k++){
                if (s.charAt(i) != s.charAt(i - k)){
                    a[i]++;
                }
                else
                    break;
            }
            if (h < a[i])h = a[i];
        }
        return h;
    }
}
